### PR TITLE
Add oauth CreateTokenFromCode API

### PR DESCRIPTION
**Pull request by ellenwei**
Based from Java reference:
OAuthCredentials:https://github.com/braintree/braintree_java/blob/master/src/main/java/com/braintreegateway/OAuthCredentials.java
OAuthCredentialsRequest:https://github.com/braintree/braintree_java/blob/master/src/main/java/com/braintreegateway/OAuthCredentialsRequest.java
OAuthGateway:https://github.com/braintree/braintree_java/blob/master/src/main/java/com/braintreegateway/OAuthGateway.java

### CODE REVIEW

File: oauth_gateway.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+type OAuthCredentialRequest struct {
+	XMLName      xml.Name `xml:"credentials"`
+	Code         string   `xml:"code"`
+	Scope        string   `xml:"scope"`
+	GrantType    string   `xml:"grantType"`
+	RefreshToken string   `xml:"refreshToken"`
+}
+
+type OAuthCredentials struct {
```
Comment (dbook13) [inline oauth_gateway.go:20]: Do we need a XML name here in order to unmarshal correctly?

**Review by dbook13 — APPROVED** [review]
Can we add a test for oauth_gateway as well?

File: oauth_gateway.go (hunk 1 above)
Comment (ellenwei) [reply oauth_gateway.go:20]: Looking at its Java API, it doesn't have a name for `OAuthCredentials` - same as the existing `PaymentMethodNonce`

**Comment by ellenwei** [comment]
> 

Same as the grant API comment, we will add integration tests for both once get the code from them.

**Pull request merged**